lib/ansible/modules/windows/win_dns_client.ps1

```diff
@@ -77,35 +79,168 @@ Function Get-NetAdapterInfo {
             $cim_params = @{
                 ClassName = "Win32_NetworkAdapterConfiguration"
                 Filter = "InterfaceIndex = $($_.InterfaceIndex)"
-                Property = "DNSServerSearchOrder", "IPEnabled"
+                Property = "DNSServerSearchOrder", "IPEnabled", "SettingID"
             }
-            $adapter_config = Get-CimInstance @cim_params
+            $adapter_config = Get-CimInstance @cim_params |
+                Add-Member -MemberType AliasProperty -Name InterfaceGuid -Value SettingID -Force -PassThru
```

Comment:
Instead of Add-Member I would do `Select-Object -Property DNSServerSearchOrder, IPEnabled, @{N='InterfaceGuid'; E={$_.SettingID}}`. `Add-Member` is slow and using `Select-Object` also forces us to make sure we aren't retrieving any extra properties that aren't specified in `$cim_params.Property`.